### PR TITLE
Update TextEdit docs - explaining 'override_selected_font_color' role

### LINKED ISSUE #38017: TextEdit: color doesn't change to font_color_selected when selecting text
**Godot version:** 3.2.1

**OS/device including version:** Windows 10, GLES3

**Issue description:**  I'm still experiencing bug #27760 even on version 3.2.1 (3.2 seemed to be affected as well), even though the fix indicated on the PR closing this bug seems to be included in the source code. (I did NOT compile the engine myself, just looked at the source code of the release.)

**Steps to reproduce:**
Change the font_color_selected (under "Custom Colors") of a TextEdit and try selecting text when running the scene. It still displays the normal font color. It's working fine for LineEdit.
OR
Change the above-mentioned property of a theme and then try to select the text in the "demo TextEdit" of the Theme editor. Same result as above.

Please let me know if this was worth opening a new issue or if there would have been some way for me to reopen the issue #27760, I'm not this familiar as this is my first-ever submitted issue on GitHub. Thanks!

### PR DESCRIPTION
**Pull request by dreamsComeTrue**
Fixes: #38017

### CODE REVIEW

File: doc/classes/TextEdit.xml [hunk 1]
```diff
@@ -433,6 +433,7 @@
 		</member>
 		<member name="mouse_default_cursor_shape" type="int" setter="set_default_cursor_shape" getter="get_default_cursor_shape" override="true" enum="Control.CursorShape" default="1" />
 		<member name="override_selected_font_color" type="bool" setter="set_override_selected_font_color" getter="is_overriding_selected_font_color" default="false">
+			If [code]true[/code], custom [code]font_color_selected[/code] will be used for selected text.
```
Comment (akien-mga) [inline doc/classes/TextEdit.xml:436]:
```suggestion
			If [code]true[/code], custom [member font_color_selected] will be used for selected text.
```

Comment (dreamsComeTrue) [reply doc/classes/TextEdit.xml:436]: Running hook: pre-commit-makerst
ERROR: Unresolved member 'font_color_selected', file: TextEdit

Comment (akien-mga) [reply doc/classes/TextEdit.xml:436]: My bad, it's `[theme_item ...]` here.

Comment (dreamsComeTrue) [reply doc/classes/TextEdit.xml:436]:
```
	If [code]true[/code], custom [theme_item font_color_selected] will be used for selected text.
```

```
Running hook: pre-commit-makerst
ERROR: Unresolved type 'theme_item font_color_selected', file: TextEdit
```
Also I couldn't find any implementation of that tag across all of the documentation, seems nobody is using it

Comment (akien-mga) [reply doc/classes/TextEdit.xml:436]: Ah right, #29868 was never finished.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.2.

**Pull request merged**